state_test.go

```diff
@@ -324,7 +326,6 @@ func TestNextStateWithInputs(t *testing.T) {
 
 	type testCase struct {
 		setupPromoter func(*testing.T, *MockPromoter)
-		setupState    func(*testing.T, *State)
```

Comment:
We now load the state from the inputs.json file for the test so there is no need to define functions to modify an existing state. This should have been a sign previously that building the next state wasn't quite as pure as I was going for.